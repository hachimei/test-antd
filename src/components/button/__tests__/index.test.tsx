import * as React from 'react'
import { Component } from "react";
import * as renderer from 'react-test-renderer';
import { mount, configure } from "enzyme";
import * as Adapter from 'enzyme-adapter-react-16';
import Button from '..';
import Icon from "../../icon";

configure({ adapter: new Adapter() });

describe('Button',() => {      
    it('renders characters correctly', () => {
      const wrapper1 = renderer.create(<Button>嘤嘤赢</Button>)
    .toJSON();
      const wrapper2 = renderer.create(<Button>嘤嘤</Button>)
    .toJSON();
    const wrapper3 = renderer.create(<Button><Icon type='book' />嘤嘤</Button>)
    .toJSON();
      expect(wrapper1).toMatchSnapshot();
      expect(wrapper2).toMatchSnapshot();
      expect(wrapper3).toMatchSnapshot();
    });

    it('have static property for type detecting',() => {
      const wrapper = mount(<Button>嘤嘤</Button>);
    // disable
      // expect(wrapper.type().__ANT_BUTTON).toBe(true);
    });

    it('should change loading state instantly by default',() => {
      class DefaultButton extends Component {
        state = {
          loading: false,
        };
        enterLoading = () => {
          this.setState({loading: true});
        }

        render(){
          return <Button loading={this.state.loading} onClick={this.enterLoading}>Button</Button>  
        }
      }
      const wrapper = mount(<DefaultButton />);
      wrapper.simulate('click');
      expect(wrapper.find('.ant-btn-loading').length).toBe(1);
    });

    it('should support link button',() => {
      const wrapper = renderer.create(
        <Button target="_blank" href="http://ant.design">link button</Button>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('fixbug renders {0} , 0 and {false}', () => {
      const wrapper = renderer.create(
        <Button>{0}</Button>
      );
      expect(wrapper).toMatchSnapshot();
      const wrapper1 = renderer.create(
        <Button>0</Button>
      );
      expect(wrapper1).toMatchSnapshot();
      const wrapper2 = renderer.create(
        <Button>{false}</Button>
      );
      expect(wrapper2).toMatchSnapshot();
    });
})