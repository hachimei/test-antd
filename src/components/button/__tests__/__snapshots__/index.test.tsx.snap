// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button fixbug renders {0} , 0 and {false} 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;

exports[`Button fixbug renders {0} , 0 and {false} 2`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    0
  </span>
</button>
`;

exports[`Button fixbug renders {0} , 0 and {false} 3`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button renders characters correctly 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    嘤嘤赢
  </span>
</button>
`;

exports[`Button renders characters correctly 2`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    嘤 嘤
  </span>
</button>
`;

exports[`Button renders characters correctly 3`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <i
    className="anticon anticon-book"
  />
  <span>
    嘤嘤
  </span>
</button>
`;

exports[`Button should support link button 1`] = `
<a
  className="ant-btn"
  href="http://ant.design"
  onClick={[Function]}
  target="_blank"
  type={undefined}
>
  <span>
    link button
  </span>
</a>
`;
